cmd/interactive.go

```diff
@@ -0,0 +1,245 @@
+		})
+	}
+
+	return adapters, nil
+}
+
+func FetchAWSRegion() ([]string, error) {
+	resp, err := http.Get(awsIPRangesURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch AWS ip-ranges.json: %w", err)
+	}
+	defer resp.Body.Close()
+
+	var data awsIPRanges
+	if err := json.NewDecoder(resp.Body).Decode(&data); err != nil {
+		return nil, fmt.Errorf("failed to decode JSON: %w", err)
+	}
+
+	regionSet := make(map[string]struct{})
+	for _, p := range data.Prefixes {
+		if p.Region != "" {
+			regionSet[p.Region] = struct{}{}
+		}
+	}
+
+	var regions []string
+	for region := range regionSet {
+		regions = append(regions, region)
+	}
+
+	return regions, nil
+}
+
+func errhandle(form *huh.Form) {
+	if err := form.Run(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func RunTui() {
+	form := huh.NewForm(
+		huh.NewGroup(
+			huh.NewSelect[string]().
+				Title("Pick a Scan Option").
+				Options(huh.NewOptions("Active Scan", "Passive Scan", "Nmap Scan", "AWS Cloud Scan", "Azure Cloud Scan")...).
+				Value(&Scantype),
+		),
+	)
+	errhandle(form)
+
+	switch {
+	case Scantype == "Active Scan":
+		pathplaceholder := CheckOsPathPlaceholder()
+		form := huh.NewForm(
+			huh.NewGroup(
+				huh.NewInput().
+					Title("Enter an export path:").
+					Placeholder(pathplaceholder).
+					Value(&Exportpath),
+			),
+		)
+		errhandle(form)
+		internal.DefaultScan()
+		header := []string{"Interface", "Dest_IP", "Dest_Mac"} //TODO: remove header for next ticket
+		internal.ActiveExport(Exportpath, header, false)
+
+	case Scantype == "Passive Scan":
+		var options []huh.Option[string]
+		pathplaceholder := CheckOsPathPlaceholder()
+		adapters, err := GetAdapters()
+		if err != nil {
+			fmt.Println("Error:", err)
+			return
+		}
+
+		if len(adapters) == 0 {
+			fmt.Println("No active adapters found.")
+			return
+		}
+		for _, a := range adapters {
+			description_transport := fmt.Sprintf("%v, %v", a.description, a.transport)
+			options = append(options, huh.Option[string]{
+				Key:   description_transport,
+				Value: a.transport,
+			})
+		}
+		form := huh.NewForm(
+			huh.NewGroup(
+				huh.NewSelect[string]().
+					Title("Select an option").
+					Options(options...).
+					Value(&SelectInterface),
+				huh.NewInput().
+					Title("Enter a duration (sec)").
+					Placeholder("default = 20").
+					Value(&durationStr).
+					Validate(func(s string) error {
+						if s == "" {
+							return nil
+						}
+						_, err := strconv.Atoi(s)
+						return err
+					}),
+				huh.NewInput().
+					Title("Enter an export path:").
+					Placeholder(pathplaceholder).
+					Value(&Exportpath),
+			),
+		)
+		errhandle(form)
+		if durationStr == "" {
+			duration = 20
+		} else {
+			duration, _ = strconv.Atoi(durationStr)
+		}
+		internal.PassiveScan(SelectInterface, duration)
+		header := []string{"Source_IP", "Protocol", "Source_MAC", "Destination_Mac", "Ethernet_Type"} //TODO: remove header
+		internal.PassiveExport(PathPassive, header)
```

Comment:
At this point, I would recommend moving the `header` variable in the `PassiveExport` function. This way, the new method signature would look like this: `internal.PassiveExport(PathPassive)` and we can define the `header` variable inside the `PassiveExport` function.